yield_plugin/src/yield_plugin.cpp

```diff
@@ -37,72 +37,381 @@ using oss = std::ostringstream;
+                                            PublishPluginDiscoveryCB plugin_discovery_publisher, 
+                                            MobilityResponseCB mobility_response_publisher)
+    : wm_(wm), config_(config), plugin_discovery_publisher_(plugin_discovery_publisher), mobility_response_publisher_(mobility_response_publisher)
   {
     plugin_discovery_msg_.name = "YieldPlugin";
     plugin_discovery_msg_.versionId = "v1.0";
     plugin_discovery_msg_.available = true;
     plugin_discovery_msg_.activated = false;
     plugin_discovery_msg_.type = cav_msgs::Plugin::TACTICAL;
-    plugin_discovery_msg_.capability = "tactical_plan/plan_trajectory";
+    plugin_discovery_msg_.capability = "tactical_plan/plan_trajectory";   
   }
 
-  bool YieldPlugin::onSpin()
+  void YieldPlugin::lookupECEFtoMapTransform()
+  {
+    tf2_listener_.reset(new tf2_ros::TransformListener(tf2_buffer_));
+    tf2_buffer_.setUsingDedicatedThread(true);
+    try
+    {
+      tf_ = tf2_buffer_.lookupTransform("earth", "map", ros::Time(0), ros::Duration(20.0)); //save to local copy of transform 20 sec timeout
+    }
+    catch (const tf2::TransformException &ex)
+    {
+      ROS_WARN("%s", ex.what());
+    }
+  }
+
+  bool YieldPlugin::onSpin() 
   {
     plugin_discovery_publisher_(plugin_discovery_msg_);
     return true;
   }
 
+  void YieldPlugin::set_lanechange_status_publisher(const ros::Publisher& publisher)
+  {
+    lanechange_status_pub_ = publisher;
+  }
+
+  std::vector<lanelet::BasicPoint2d> YieldPlugin::detect_trajectories_intersection(std::vector<lanelet::BasicPoint2d> trajectory1, std::vector<lanelet::BasicPoint2d> trajectory2) const
+  {
+    // the vectors are converted to linestrings so their intersection point can be calculated
+
+    // A linestring is a Curve with linear interpolation between Points.
+    boost::geometry::model::linestring<lanelet::BasicPoint2d> traj1;
+    boost::geometry::model::linestring<lanelet::BasicPoint2d> traj2;
+    for (auto tpp1:trajectory1)
+    {
+      boost::geometry::append(traj1, tpp1);
+    }
+
+    for (auto tpp2:trajectory2)
+    {
+      boost::geometry::append(traj2, tpp2);
+    }
+
+    std::vector<lanelet::BasicPoint2d> intersection_points;
+    boost::geometry::intersection(traj1, traj2, intersection_points);
+
+    return intersection_points;
+  }
+
+  std::vector<lanelet::BasicPoint2d> YieldPlugin::detect_trajectories_intersection2(std::vector<lanelet::BasicPoint2d> self_trajectory, std::vector<lanelet::BasicPoint2d> incoming_trajectory) const
+  {
+    std::vector<lanelet::BasicPoint2d> intersection_points;
+    boost::geometry::model::linestring<lanelet::BasicPoint2d> self_traj;
+    for (auto tpp:self_trajectory)
+    {
+      boost::geometry::append(self_traj, tpp);
+    }
+    // distance to consider trajectories colliding (chosen based on lane width and vehicle size)
+    double acceptable_distance = 2.0;
```

Comment:
We should definitely parameterize this as a ROS param. The bounding box (or I suppose circle here) size will likely be an important tuning parameter for this scenario. I would probably double this as a default value as well (or more) since it needs to account for sufficient space for both vehicles and since our vehicles are longer than they are wide.